src/compiler/checker.ts

```diff
@@ -11433,8 +11429,7 @@ namespace ts {
 
                 let result = Ternary.False;
                 const saveErrorInfo = errorInfo;
-                const saveIsIntersectionConstituent = isIntersectionConstituent;
-                isIntersectionConstituent = false;
+                let isIntersectionConstituent = !!isApparentIntersectionConstituent;

```

Comment:
I think the new distinction between `isIntersectionConstituent` and `isApparentIntersectionConstituent` is that `isIntersection` is not enclosed in checkTypeRelated, so it only propagates through passing to recursive calls. And `isApparentIntersectionConstituent` *is* enclosed in checkTypeRelated, but doesn't pass recursively.

Is that accurate? The distinction is subtle and warrants some explanation and/or renaming of variables because it'll be hard to reconstruct the design by reading the code later.

Reply:
`isIntersectionConstituent` "propagates" through all calls, but is reset on every visit of `isRelatedTo`, while `isApparentIntersectionConstituent` is an argument used to prevent that reset